kernel/queue.c

```diff
@@ -57,12 +57,14 @@ void k_queue_init(struct k_queue *queue)
 	SYS_TRACING_OBJ_INIT(k_queue, queue);
 }
 
+#if !defined(CONFIG_POLL)
```

Comment:
This is simpler to start with, if we do choose to go with mandatory CONFIG_POLL there is no reason to keep the logic around wait_q but many objects that do have wait_q don't have POLL_EVENT yet so this will serve as test bed how to convert the objects so we can take advantage of k_poll if that became built-in. I actually asked about this making k_poll part of the always enabled set of APIs in the mailing list but nobody responded.